pkg/operator/ceph/controllers/nodedrain/add.go

```diff
@@ -0,0 +1,65 @@
+package nodedrain
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/rook/rook/pkg/operator/ceph/controllers/controllerconfig"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	minStoreResyncPeriod = 10 * time.Hour
+	controllerName       = "nodedrain-controller"
+	drainDetectionLabel  = "rook-ceph-canary"
+)
+
+// Add adds a new Controller based on nodedrain.ReconcileNode that reconciles nodes
+// by creating owned drain detection deployment tied to that node by nodeSelector.
+// Other controllers can watch the resultant pod by it's node label and
+func Add(mgr manager.Manager, opts *controllerconfig.Options) error {
+	reconcileNode := &ReconcileNode{
+		client:  mgr.GetClient(),
+		scheme:  mgr.GetScheme(),
+		options: opts,
+	}
+	reconciler := reconcile.Reconciler(reconcileNode)
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return err
```

Comment:
please a more meaningful error message like `fmt.Errorf("failed to create new nodedrain controller: %+v", err)`.

Reply:
The controller error has that info